skipruntime-ts/helpers/src/external.ts

```diff
@@ -104,6 +113,9 @@ export class Polled<S extends Json, K extends Json, V extends Json>
     private readonly url: string,
     private readonly duration: number,
     private readonly conv: (data: S) => Entry<K, V>[],
+    private readonly encodeParams: (params: {
+      [param: string]: Json;
+    }) => string = defaultParamEncoder,
```

Comment:
This generalization is specific to `Polled` rather than appearing in `ExternalResource` because with `Polled` the user doesn't get to define `open`, while in the generic case they can implement whatever param encoding they need, right?

Reply:
Yes, I tried to push this down in the class hierarchy so that it doesn't constrain anything for the generic case -- in which you just have to know how to "open" and "close" with the given Json params.

I suppose there's an argument to lift it up to the ExternalService though, to avoid repetition if you have multiple ExternalResources on the same service, all using the same param encoding...

Reply:
I don't have an intuition about repetition over multiple ExternalResources, but guess that surely people can factor out their implementations to dedup that if needed. So I'd leave it as you have it for now.